mmdet/evaluation/metrics/openimages_metric.py

```diff
@@ -1,237 +1,185 @@
 # Copyright (c) OpenMMLab. All rights reserved.
-import copy
-from collections import OrderedDict
-from typing import List, Optional, Sequence, Union
+from typing import List, Optional, Sequence, Tuple, Union
 
 import numpy as np
-from mmengine.evaluator import BaseMetric
-from mmengine.logging import MMLogger, print_log
+from mmengine.logging import print_log
+from mmeval import OIDMeanAP
+from terminaltables import AsciiTable
 
 from mmdet.registry import METRICS
-from ..functional import eval_map
 
 
 @METRICS.register_module()
-class OpenImagesMetric(BaseMetric):
-    """OpenImages evaluation metric.
+class OpenImagesMetric(OIDMeanAP):
+    """A wrapper of :class:`mmeval.OIDMeanAP`.
 
-    Evaluate detection mAP for OpenImages. Please refer to
-    https://storage.googleapis.com/openimages/web/evaluation.html for more
-    details.
+    This wrapper implements the `process` method that parses predictions and
+    labels from inputs. This enables ``mmengine.Evaluator`` to handle the data
+    flow of different tasks through a unified interface.
+
+    In addition, this wrapper also implements the ``evaluate`` method that
+    parses metric results and print pretty table of metrics per class.
 
     Args:
-        iou_thrs (float or List[float]): IoU threshold. Defaults to 0.5.
-        ioa_thrs (float or List[float]): IoA threshold. Defaults to 0.5.
+        iof_thrs (float ｜ List[float]): IoF thresholds. Defaults to 0.5.
```

Comment:
This will cause BC-breaking. Can we avoid this problem?

Reply:
In my opinion, refactoring to support MMEval is also a big BC-breaking

We can add warning such as:

```
ioa_thrs = kwargs.pop('ioa_thrs', None)
if ioa_thrs is not None and 'iof_thrs' not in kwargs:
    kwargs['iof_thrs'] = ioa_thrs
    warnings.warn(
        'DeprecationWarning: The `ioa_thrs` parameter of '
         '`OpenImagesMetric` is deprecated, use `iof_thrs` instead!')
```